src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -59,8 +60,33 @@ public static void ToSerializeCall(this CodeWriter writer, JsonSerialization ser
 
                     foreach (JsonPropertySerialization property in obj.Properties)
                     {
-                        bool hasNullableType = property.Property != null && property.Property.Declaration.Type.IsNullable;
-                        using (hasNullableType ? writer.Scope($"if ({property.Property!.Declaration.Name} != null)") : default)
+                        bool hasNullableType = property.Property?.Declaration.Type.IsNullable == true;
+                        bool emptyAsUndefined = property.Property?.EmptyAsUndefined == true;
+
+                        CodeWriter.CodeWriterScope? scope = default;
+                        if (hasNullableType || emptyAsUndefined)
+                        {
+                            var propertyName = property.Property!.Declaration.Name;
+                            writer.Append($"if (");
+                            if (hasNullableType)
+                            {
+                                writer.Append($"{propertyName} != null");
+                            }
+
+                            if (emptyAsUndefined)
+                            {
+                                if (hasNullableType)
+                                {
+                                    writer.Append($" && ");
+                                }
+                                writer.UseNamespace(typeof(Enumerable).Namespace!);
+                                writer.Append($"{propertyName}.Any()");
```

Comment:
I recommend `Count` if you know it's `IList<T>` or `IReadOnlyList<T>`. `Any()` in .NET Framework (as opposed to .NET Core) is slower and allocates an enumerator (creates enumerator, calls MoveNext() once).